internal/admission/validator.go

```diff
@@ -63,20 +64,25 @@ func (validator KongHTTPValidator) ValidatePlugin(
 		return false, "plugin name cannot be empty", nil
 	}
 	var plugin kong.Plugin
+	var configuration kong.Configuration
 	plugin.Name = kong.String(k8sPlugin.PluginName)
-	if k8sPlugin.Config != nil {
-		plugin.Config = kong.Configuration(k8sPlugin.Config)
+	if k8sPlugin.Config.Raw != nil {
+		err := json.Unmarshal(k8sPlugin.Config.Raw, &configuration)
+		if err != nil {
+			return false, "", fmt.Errorf("could not unmarshal plugin configuration: %s", err)
+		}
+		plugin.Config = configuration
```

Comment:
All cases here (`Raw` being `nil`, `Raw` being `make([]byte, 10)`, `Raw` being invalid JSON, `Raw` being valid JSON) need to be tested.

Reply:
8dd7a06 for what I think we can cover. That omits:

- More than one invalid JSON case. These are all equivalent, so `make([]byte, 10)` isn't any different from `{{}`. They all fail `json.Unmarshal()`.
- Valid JSON/success cases in general. The webhook relies on a live Kong instance to bounce `POST /schemas/plugins/validate` requests off of, so we can't unit test anything that it cannot validate independently.
- `Raw` being `nil` (empty, rather, but whatever). This is part of the above: an empty configuration is often valid, since plugins will use default settings if none are provided. When it isn't valid (e.g. OIDC's required `issuer`, which has no default), we need the validate request to tell us.

Reply:
(blocking)

> More than one invalid JSON case. These are all equivalent, so make([]byte, 10) isn't any different from {{}. They all fail json.Unmarshal().

A `nil` slice and a `make([]byte, 0, 10)` slice both have length 0, but they cause execution to follow different code paths in this implementation. This, I think, justifies the need for a separate test case per se.

> Valid JSON/success cases in general. The webhook relies on a live Kong instance to bounce POST /schemas/plugins/validate requests off of, so we can't unit test anything that it cannot validate independently.

Please correct me if I'm wrong, but the main code path (that performs the request to the endpoint you mentioned) in `ValidatePlugin` is untested. To me, this justifies blocking this change until a test gets written.

This test will require mocking out `kong.Client` in `KongHTTPValidator` (replacing the `Client *kong.Client` field with an interface value). On an upside, this will set a good example towards fixing https://github.com/Kong/deck/issues/224.

Reply:
A mock test here has little use, since `/schemas/plugins/validate` is an oracle by design. We don't have knowledge of individual plugin schemas in the webhook, and rely on Kong to provide that information. Any test that takes Kong out of the picture will need to present made-up responses. We'd get coverage of the code path, but tests wouldn't reflect any actual behavior of the oracle for a given input.

I don't think that should block this in any case. This PR doesn't change the functionality of the webhook, it only updates it to handle the different type. Building out a mock implementation of `kong.Client` isn't really in scope for making KongPlugin compatible with kubebuilder, which is what we're trying to do here.